src/agent0/core/hyperdrive/crash_report/known_error_checks.py

```diff
@@ -154,6 +155,91 @@ def check_for_invalid_balance(trade_result: TradeResult, interface: HyperdriveRe
     return trade_result
 
 
+def check_for_insufficient_allowance(trade_result: TradeResult, interface: HyperdriveReadInterface) -> TradeResult:
+    """Detects insufficient allowance in trade_result and adds additional information to the
+    exception in trade_result.
+
+    Arguments
+    ---------
+    trade_result: TradeResult
+        The trade result object from trading.
+    interface: HyperdriveReadInterface
+        The hyperdrive read interface to compute expected balances for trades.
+
+    Returns
+    -------
+    TradeResult
+        A modified trade_result that has a custom exception argument message prepended
+    """
+    assert trade_result.account is not None
+    agent_address = trade_result.account.address
+    assert trade_result.trade_object is not None
+    trade_type = trade_result.trade_object.market_action.action_type
+    trade_amount = trade_result.trade_object.market_action.trade_amount
+
+    base_token_contract_address = interface.base_token_contract.address
+    hyperdrive_contract_address = interface.hyperdrive_contract.address
+
+    insufficient_allowance = False
+    add_arg = None
+    match trade_type:
+        case HyperdriveActionType.INITIALIZE_MARKET:
+            raise ValueError(f"{trade_type} not supported!")
```

Comment:
Why an error here? If there is no "approval" involved in initialize then I think the move is to just safely return the `TradeResult` object. That way I don't have to be as careful about running this check on my trade object.

Reply:
or I guess why is this different from the case with closing & removing?

Reply:
We never support doing an initialize trade, but `INITIALIZE_MARKET` is a valid enum, hence it's just an extra check to ensure this isn't a valid action.